### PR TITLE
:seedling: bump catalogd to v0.11.0

### PR DESCRIPTION
**Pull request by everettraven**
# Description
- Bumps catalogd --> v0.11.0
- Bumps kind to v0.20.0 and ensures we are using a 1.28 node image for testing

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -25,6 +25,7 @@ export ARTIFACT_PATH ?=
 
 OPERATOR_CONTROLLER_NAMESPACE ?= operator-controller-system
 KIND_CLUSTER_NAME ?= operator-controller
+KIND_CLUSTER_IMAGE ?= kindest/node:v1.28.0@sha256:b7a4cad12c197af3ba43202d3efe03246b3f0793f162afb40a33c923952d5b31
```
Comment (joelanford) [inline Makefile:28]: Why hardcode this and not take the default from the version of `kind` we're using?

Comment (everettraven) [reply Makefile:28]: The latest version of kind has a default node image based on 1.27.3

Comment (everettraven) [reply Makefile:28]: https://github.com/kubernetes-sigs/kind/releases/tag/v0.20.0

Comment (joelanford) [reply Makefile:28]: Would it be possible to use `ENVTEST_VERSION` to derive a kind image ref?

Setting a hardcoded digest strikes me as something that will rot, and we'll forget about.

Alternately, maybe we just accept that `kind`'s default might be off by a release or two?

Comment (everettraven) [reply Makefile]: https://github.com/operator-framework/operator-controller/pull/578/commits/2d1e78b99c5fc2a1c19b0c009cb591d9f86e5c65 ?

**Review by grokspawn — APPROVED** [review]

**Pull request merged**